### PR TITLE
[blockindex] remove DeleteTipBlock()

### LINKED ISSUE #2645: remove DeleteTipBlock() from Indexer interface
**What would you like to be added**:
remove DeleteTipBlock() from Indexer (in indexer.go) and BlockIndexer (in blockdao.go) interface

**Why is this needed**:
for blockdao DB, you can delete the tip block. But for indexer, such operation is not very meaningful and maybe impossible (for statedb/factory)

### PR DESCRIPTION
**Pull request by dustinxie**
# Description
for most block indexers, it is not possible to remove the tip block just being added

Fixes #2645 

## Type of change
Please delete options that are not relevant.
- [] Bug fix (non-breaking change which fixes an issue)
- [] New feature (non-breaking change which adds functionality)
- [x] Code refactor or improvement
- [] Breaking change (fix or feature that would cause a new or changed behavior of existing functionality)
- [] This change requires a documentation update

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [] fullsync
- [] Other test (please specify)

**Test Configuration**:
- Firmware version:
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by Liuhaai** [comment]
What is the purpose of deleting the interface in the pr? This issue has been discussed before at https://github.com/iotexproject/iotex-core/pull/2647

File: blockchain/blockdao/blockdao.go [hunk 1]
```diff
@@ -272,31 +272,10 @@ func (dao *blockDAO) DeleteBlockToTarget(targetHeight uint64) error {
...
-		// delete block index if there's indexer
-		for _, indexer := range dao.indexers {
-			if err := indexer.DeleteTipBlock(ctx, blk); err != nil {
-				return err
-			}
-		}
-
 		if err := dao.blockStore.DeleteTipBlock(); err != nil {
 			return err
 		}
```
[the remark this answers is not in the document]
Comment (dustinxie) [reply blockchain/blockdao/blockdao.go:278]: yes, we can have another PR to also remove this API from chain db

**Comment by dustinxie** [comment]
> What is the purpose of deleting the interface in the pr? This issue has been discussed before at #2647

oh, #2647 already discussed this. The purpose is that `DeleteTipBlock()` is a wrong API for indexer (see #2645), and it should be removed from the code

**Review by envestcc — APPROVED** [review]

**Comment by raullenchai** [comment]
Close b/c it has been inactive for 1+yr.

**Pull request closed without merging**